azext_edge/edge/providers/check/base.py

```diff
@@ -785,29 +785,43 @@ def process_dict_resource(
                 padding=padding + PADDING_SIZE
             )
         else:
-            display_text = ""
+            display_text = f"{key}: "
+            value_padding = padding
             if isinstance(value, str) and len(value) > 50:
-                display_text = f"{key}:"
-                check_manager.add_display(
-                    target_name=target_name,
-                    namespace=namespace,
-                    display=Padding(display_text, (0, 0, 0, padding))
-                )
-                display_text = f"[cyan]{value}[/cyan]"
-                check_manager.add_display(
-                    target_name=target_name,
-                    namespace=namespace,
-                    display=Padding(display_text, (0, 0, 0, padding + PADDING_SIZE))
-                )
-            else:
-                # replace empty string with N/A
-                value = value if value else "N/A"
-                display_text = f"{key}: [cyan]{value}[/cyan]"
                 check_manager.add_display(
                     target_name=target_name,
                     namespace=namespace,
                     display=Padding(display_text, (0, 0, 0, padding))
                 )
+                value_padding += PADDING_SIZE
+                display_text = ""
+            display_text += process_value_color(
+                check_manager=check_manager,
+                target_name=target_name,
+                key=key,
+                value=value
+            )
+            check_manager.add_display(
+                target_name=target_name,
+                namespace=namespace,
+                display=Padding(display_text, (0, 0, 0, value_padding))
+            )
+
+
+def process_value_color(
```

Comment:
The only concern I can think of so far for report error when key contains "error" text in `process_dict_resource` is: would there be possibility that key contains "error" that is not actually error? for example if there is a kvp like `"error_logging_disabled": True`. But again, this is a good solution for capturing nested error since we never know when error can happen and in **this specific case**, there is no hint in CRD that an error can happen here

Reply:
For now, we can assume that if there is an error in the key, then it is talking about an error message more often than not. We can always add in exceptions (ex: value is NoError means that this is not an error).

I have made this into a helper function so it is easier to track and edit.